### PR TITLE
Flesh out Path APIs

### PR DESCRIPTION
**Pull request by armanbilge**

### CODE REVIEW

File: io/shared/src/main/scala/fs2/io/file/PathApi.scala [hunk 1]
```diff
@@ -23,13 +23,46 @@ package fs2
...
+
+  def resolve(path: Path): Path
+  
   def normalize: Path
+
+  def isAbsolute: Boolean
+
+  def absolute: Path
+  
+  def names: Seq[Path]
```
Comment (mpilquist) [inline io/shared/src/main/scala/fs2/io/file/PathApi.scala:47]: s/Seq/List/ so we have type class instances available

Comment (armanbilge) [reply io/shared/src/main/scala/fs2/io/file/PathApi.scala:47]: I thought they got added for list?

Comment (armanbilge) [reply io/shared/src/main/scala/fs2/io/file/PathApi.scala:47]: https://github.com/typelevel/cats/blob/main/core/src/main/scala/cats/instances/seq.scala

Comment (mpilquist) [reply io/shared/src/main/scala/fs2/io/file/PathApi.scala:47]: Okay but still should be List - Seq offers no advantage here

**Pull request merged**